data_handler/handlers/liquidable_debt/debt_handlers.py

```diff
@@ -49,45 +76,189 @@ def prepare_data(self, protocol_name: str, path: str = LOCAL_STORAGE_PATH) -> di
+        Data to calculate liquidable debt for:
+        :param data: {
+            Row number: {
+                Field name: Token value
+                 or Decimal - Field value (the price as example)
+                 or {
+                    Token name: Token value
+                }
+            }
+        }
         :return: A dictionary of the ready liquidable debt data.
         """
-        result_data = deepcopy(data)
+        # FIXME Read loan_entities from the database
+        # FIXME concentrate for one Protocol and one collateral_token and debt_token
+        # FIXME go through all data and init them as loan_entities
+        result_data = dict()
+        current_prices = Prices()
+        asyncio.run(current_prices.get_lp_token_prices())
+        INTEREST_RATE_MODELS_MAPPING = {
+            protocol.value.lower(): self.get_interest_rate_models_data(protocol_name=protocol.value)
+            for protocol in AvailableProtocolID
+        }
 
-        for row_number in data:
-            state = self.state_class(verbose_user=data[row_number][USER_FIELD_NAME])
+        # init just once for improving performance
+        state = self.state_class()
+        # TODO: 1. Read data from db for state.loan_entities from LoanState model
+        # TODO: 2. Set last interest rate record by block
+        # state.interest_rate_models = InterestRateModels(..)
+        # # Set up collateral and debt interest rate models
+        # state.collateral_interest_rate_models = InterestRateModels(
+        #     values=interest_rate_models[COLLATERAL_FIELD_NAME.lower()]
+        # )
+        # state.debt_interest_rate_models = InterestRateModels(
+        #     values=interest_rate_models[DEBT_FIELD_NAME.lower()]
+        # )
+        # for index, row in data.items():
+            # even if it isn't per-user data, we need to provide a user ID
+            # so like that we're able to provide debt and collateral values
+            # TODO use just one collateral token and debt token
+            # for collateral_token in row[COLLATERAL_FIELD_NAME].keys():
+            #     hypothetical_collateral_token_prices = self.get_prices_range(
+            #         collateral_token_name=collateral_token,
+            #         current_price=current_prices.prices.values[collateral_token]
+            #     )
 
-            state.loan_entities[data[row_number][USER_FIELD_NAME]].debt.values = {
-                key: value
-                for key, value in data[row_number][DEBT_FIELD_NAME].items()
-            }
-            state.loan_entities[data[row_number][USER_FIELD_NAME]].collateral.values = {
-                key: value
-                for key, value in data[row_number][COLLATERAL_FIELD_NAME].items()
-            }
+        # Go through first hypothetical prices and then through the debts
+        for hypothetical_price in hypothetical_collateral_token_prices:
+            current_prices.prices.values[collateral_token] = hypothetical_price
```

Comment:
Don't do this now, this step is done inside `state.compute_liquidable_debt_at_price` later.